tests/test_xmos_calendar.py

```diff
@@ -0,0 +1,130 @@
+from unittest import TestCase
+
+import pandas as pd
+from pytz import UTC
+
+from trading_calendars.exchange_calendar_xmos import XMOSExchangeCalendar
+
+from .test_trading_calendar import ExchangeCalendarTestBase
+
+
+class XMOSCalendarTestCase(ExchangeCalendarTestBase, TestCase):
+
+    answer_key_filename = 'xmos'
+    calendar_class = XMOSExchangeCalendar
+
+    # The XMOS is open from 10:00AM to 6:45PM.
+    MAX_SESSION_HOURS = 8.75
+
+    # In 2009 in Russia, daylight savings began on March 29th and ended on
+    # October 25th.
+    DAYLIGHT_SAVINGS_DATES = ['2009-03-30', '2009-10-26']
+
+    HAVE_EARLY_CLOSES = False
+
+    # The default `MINUTE_INDEX_TO_SESSION_LABELS_START` is not a session in
+    # the XMOS calendar so overwrite that date here.
+    MINUTE_INDEX_TO_SESSION_LABELS_START = pd.Timestamp('2019-01-04', tz=UTC)
+    MINUTE_INDEX_TO_SESSION_LABELS_END = pd.Timestamp('2019-04-04', tz=UTC)
+
+    # None of the dates in the default range for these are trading days in the
+    # XMOS, so overwrite them here.
+    TEST_START_END_FIRST = pd.Timestamp('2019-01-13', tz=UTC)
+    TEST_START_END_LAST = pd.Timestamp('2019-01-20', tz=UTC)
+    TEST_START_END_EXPECTED_FIRST = pd.Timestamp('2019-01-14', tz=UTC)
+    TEST_START_END_EXPECTED_LAST = pd.Timestamp('2019-01-18', tz=UTC)
```

Comment:
I don't think I've been checking that these are valid dates for previous calendars... would there be a failed test if the default was not changed but fell on non-trading days?  If not, I should probably go back and make sure these are valid dates for the calendars I've done.

Reply:
> would there be a failed test if the default was not changed but fell on non-trading days?

Yes, one of [these checks](https://github.com/quantopian/trading_calendars/blob/master/tests/test_trading_calendar.py#L828-L835) would fail. Which means if all of your previous calendars are passing this test you shouldn't need to change anything.

In my case, I was actually hitting an error [on this line](https://github.com/quantopian/trading_calendars/blob/master/tests/test_trading_calendar.py#L826) during construction of the calendar, even before those assertions, because there were no trading days in the default range _at all_.

Reply:
Oh okay, noted!